client/src/appsupport.res

```diff
@@ -0,0 +1,310 @@
+    let records: array<Dom.mutationRecord> = observer->MutationObserver.takeRecords
+    records|>Js.Array.forEach(record=>{
+      record->removedNodes
+      |> Webapi.Dom.NodeList.toArray
+      |> Js.Array.forEach(node=>{
+        Js.log(node)
+        // if (node["id"]== "unsupportedBrowser") {
+        //   reload := true
+        //   }
+      })
+      if (reload.contents){
+          Webapi.Dom.location->Webapi.Dom.Location.reload}
+    })
+    observer->MutationObserver.observe(
+      document->Document.documentElement,
+      {
+        "subtree": true,
+        "childList": true,
+        "attributes": true,
+        "attributeOldValue": true,
+        "characterData": true,
+      },
+    )
+    }
+  )}
+
+
+
+
+// ---------------------------
+// Analytics
+// ---------------------------
+
+
+// ---------------------------
+// Allows us capture certain keys and stop them from affecting the browser.
+// ---------------------------
+
+
+
+let stopKeys =  event =>{
+  open! KeyboardEvent
+  // Don't ever attempt to save the HTML of the page.
+ if (ctrlKey(event)|| metaKey(event)) && key(event)=="s"{
+  event -> preventDefault
+ }
+
+ // `Ctrl-K` is meant to open the omnibox on Linux, but without this preventDefault
+ // it will focus the browser's URL bar after creating the omnibox.
+
+  if ctrlKey(event)&& key(event)=="k" && Entry.getBrowserPlatform() == Linux{
+      event -> preventDefault
+  }
+}
+
+
+// ---------------------------
+// Is it chrome?
+// ---------------------------
+
+let getBrowserPlatform = () => {
+  // Checks if mac
+  let isMac = Entry.getBrowserPlatform() == Mac
+  // Check if Linux
+  let isLinux = Entry.getBrowserPlatform() == Linux
+  // Check if Windows
+  let isWindows = Entry.getBrowserPlatform() == Windows
+  // known platform
+  if (isMac) {
+    0
+  } else if (isLinux) {
+    1
+  } else if (isWindows) {
+    2
+  } else {
+    3
+  }
+}
+
+// ---------------------------
+// Rollbar
+// ---------------------------
+
+module Rollbar = {
+  type person ={
+    id: string,
+    username: string,
+  }
+
+  type rollbarConfig = {
+    enabled: bool,
+    payload: person,
+  }
+  type rollbar =
+  {
+  init: rollbarConfig => unit,
+  error: Js.Json.t => unit,
+  configure: rollbarConfig => unit,
+  }
+  @module external rollbar: rollbar = "rollbar"
+  let init = (rollbarConfig: rollbarConfig) => {
+    rollbar.init(rollbarConfig)
+  }
+  let error = (error: Js.Json.t) => {
+    rollbar.error(error)
+  }
+  let payload = (payload: person) => {
+    rollbar.configure({enabled: true, payload: payload})
+      }
+  let enabled = (enabled: bool) => {
+    rollbar.configure({enabled: enabled, payload: {id: "", username: ""}})
+  }
+  let configure = ({enabled: enabled, payload: payload}) => {
+    rollbar.configure({enabled: enabled, payload: payload})
+  }
+
+let username=""
+let userID=""
+  let rollbarConfig = {
+    enabled: true,
+    payload: { id: userID, username: username }
```

Comment:
I am not sure where to get username and userID from. Used empty strings just to pass this step.

Reply:
Those are created in UI.html: https://github.com/darklang/dark/blob/main/backend/src/ApiServer/Templates/ui.html#L106

Reply:
> I am not sure where to get username and userID from.

based on Paul's link, they should just 'be available' to you in `appsupport.js` - the values are 'injected' when you load the page, from `backend/src/ApiServer/Ui.fs` into `ui.html`, such that both `username` and `userID` should be global variables. If you're not seeing values for them, something may be wrong.

Reply:
They are available in `appsupport.js`. I need them in `appsupport.res` (the Rescript version of appsupport.js). 
Still not sure if I am supposed to create this file (appsupport.res) or if I should move the functions somewhere else.

Reply:
Look at client/workers/Fetcher.res - I think you can use the pattern of

`@val external buildHash: string = "buildHash"`

to read those globals and pull them into ReScript world?

Reply:
e.g. add  `@val external username: string = "username"` to the top of `appsupport.res`, maybe?

Reply:
> Still not sure if I am supposed to create this file (appsupport.res) or if I should move the functions somewhere else.
I'm not sure (depends on the fns) but moving into an appsupport.res seems a great step forward. Pulling the fns into other files can be in a follow-up PR, I think.

Reply:
> e.g. add `@val external username: string = "username"` to the top of `appsupport.res`, maybe?

This worked and will enable me to convert more functions. Thank you so much! (I still need to test all of these functions to ensure they work properly.)